query/influxql/transpiler.go

```diff
@@ -691,27 +664,75 @@ func (t *transpilerState) from(m *influxql.Measurement) (ast.Expression, error)
 							Name: "bucket",
 						},
 						Value: &ast.StringLiteral{
-							Value: fmt.Sprintf("%s/%s", db, rp),
+							Value: t.config.Bucket,
 						},
 					},
 				},
 			},
 		}
 	} else {
-		// use mapping bucket id
-		args = []ast.Expression{
-			&ast.ObjectExpression{
-				Properties: []*ast.Property{
-					{
-						Key: &ast.Identifier{
-							Name: "bucketID",
+		if t.dbrpMappingSvc == nil {
+			return nil, errors.New("cannot transpiler, no dbrp mappings provided")
+		}
+		db, rp := m.Database, m.RetentionPolicy
+		if db == "" {
+			if t.config.DefaultDatabase == "" {
+				return nil, errors.New("database is required")
```

Comment:
What do you think of also giving guidance in the error message to add db to the query?

Reply:
Currently the only way this path is hit is if goreplay fails to send the appropriate parameters. Normally yes informing the user would be good, in this case its an internal code path.